Serbian Cyrillic pre-translated validation messages

sttewan opened the pull request:
New sr_RS/Zend_Validate.php for Serbian Cyrillic pre-translated validation messages.

Commit:
Create Zend_Validate.php

Creating new sr_RS/Zend_Validate.php for Serbian Cyrillic pre-translated validation messages.

Commit:
Merge pull request #1 from sttewan/local-dev

Create Zend_Validate.php

Commit:
Update Zend_Validate.php

Updating sr_RS/Zend_Validate.php for Serbian Cyrillic pre-translated validation messages.

Commit:
Merge pull request #2 from sttewan/local-dev

Update Zend_Validate.php

Commit:
Create Zend_Captcha.php

Creating sr_RS/Zend_Captcha.php for Serbian Cyrillic pre-translated validation messages.

Commit:
Merge pull request #3 from sttewan/local-dev

Create Zend_Captcha.php

ManInTheBox commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+return array(
+    // Zend\Authentication\Validator\Authentication
+    "Invalid identity" => "Погрешан идентитет",
+    "Identity is ambiguous" => "Идентитет је сумњив",
```
Not sure about this:

``` diff
- сумњив
+ двосмислен
```

@nikolaposa thoughts?

sttewan replied:
"двосмислен" is OK

nikolaposa replied:
"dvosmislen", "nejasan"...

sttewan replied:
Changed to "двосмислен".

ManInTheBox commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+    "Authentication failed" => "Неуспела аутентификација",
+
+    // Zend\I18n\Validator\Alnum
+    "Invalid type given. String, integer or float expected" => "Погрешан тип унет. Очекиван је текст, цео број или стварни број",
```
I'd rather use native english words which are already known by the developers: `string`, `integer`, `float`. I think this phrase is used in validation context so it just makes more sense to use _programming_ terms.

lahaxearnaud replied:
+1

ManInTheBox replied:
Or I might be wrong and this message is displayed to end user. In that case your solution would be better.

sttewan replied:
From the end users perspective it is easier to understand translated types.

umpirsky replied:
+1 for translated types.

sttewan replied:
Changed to developers perspective.

ManInTheBox commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+    // Zend\I18n\Validator\Alnum
+    "Invalid type given. String, integer or float expected" => "Погрешан тип унет. Очекиван је текст, цео број или стварни број",
+    "The input contains characters which are non alphabetic and no digits" => "Унос садржи карактере који нису алфабетски и нису бројеви",
+    "The input is an empty string" => "Унет је празан текст",
```
Same thing as I described [above](https://github.com/zendframework/zend-i18n-resources/pull/11/files#r43244092) though this particular one could be easily understood by the devs.

sttewan replied:
So, it will be end user or developer?

sttewan replied:
Changed to developers perspective.

ManInTheBox commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+
+    // Zend\I18n\Validator\DateTime
+    "Invalid type given. String expected" => "Погрешан тип унет. Очекиван је текст",
+    "The input does not appear to be a valid datetime" => "Унет је неисправан датум и време",
```
Hmm I think that `datetime` here is particular data type (expects well-known format). Your translation is _"date and time"_. Again, I'm all in favor of keeping original to avoid misunderstanding: `datetime`.

sttewan replied:
Ok, datetime then.

ManInTheBox commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+    "Invalid type given. String expected" => "Погрешан тип унет. Очекиван је текст",
+
+    // Zend\Validator\Between
+    "The input is not between '%min%' and '%max%', inclusively" => "Унос није између '%min%' и '%max%', инклузивно",
```
``` diff
+ Унос није искључиво између '%min%' и '%max%'
```

sttewan replied:
Your translation is better.

ManInTheBox commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+
+    // Zend\Validator\Between
+    "The input is not between '%min%' and '%max%', inclusively" => "Унос није између '%min%' и '%max%', инклузивно",
+    "The input is not strictly between '%min%' and '%max%'" => "Унос није искључиво између '%min%' и '%max%'",
```
Well, not sure: `стриктно`?

sttewan replied:
Yes, it's better.

ManInTheBox commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+    "No token was provided to match against" => "Токен за проверу није унет",
+
+    // Zend\Validator\InArray
+    "The input was not found in the haystack" => "Унос није пронађен у низу",
```
Your solution is fine although `haystack` term is important.

sttewan replied:
haystack then, no translation

ManInTheBox replied:
I think both are good. Usually `haystack` just explicitly defines array. It might not be worth updating this translation.

sttewan replied:
That's why I translated it as array. Пласт is not so good :)

nikolaposa replied:
"Унос није пронађен у захтеваном низу вредности"?

sttewan replied:
Agree with @nikolaposa. This is changed.

ManInTheBox commented:
:+1:

umpirsky commented on languages/sr_RS/Zend_Captcha.php:
```diff
@@ -0,0 +1,23 @@
+return array(
+    // Zend\Captcha\ReCaptcha
+    "Missing captcha fields" => "Недостају captcha поља",
+    "Failed to validate captcha" => "Неуспела captcha валидација",
```
Looks like we imported this word as well https://sr.wikipedia.org/wiki/%D0%9A%D0%B0%D0%BF%D1%87%D0%B0.

ManInTheBox replied:
Wut!? Rather NO in my opinion :D

nikolaposa replied:
Agree with @ManInTheBox, you simply don't want to translate some words, i.e. web, captcha, server, framework (frejmvork), etc. :)

umpirsky replied:
@nikolaposa Like file = датотека. :)

sttewan replied:
Nothing to change here.

umpirsky commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+    // Zend\Validator\Barcode
+    "The input failed checksum validation" => "Унос није прошао проверу вредности",
+    "The input contains invalid characters" => "Унос садржи неисправне карактере",
+    "The input should have a length of %length% characters" => "Унос треба да има дужину од %length% карактера",
```
What if length is 1? Should be `Унос треба да има дужину од %length% карактер` in that case. [Here](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Validator/Resources/translations/validators.sr_Cyrl.xlf#L87) is how we handle it in Symfony.

sttewan replied:
You are right. This was overlooked. Something like this is better "Дозвољени број карактера је %length%".

umpirsky commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+    "One or more files can not be read" => "Један или више фајлова су нечитљиви",
+
+    // Zend\Validator\File\Hash
+    "File does not match the given hashes" => "Фајл не одговара датим hash вредностима",
```
`hash` => `хеш`

sttewan replied:
I would not change it, same as for the other examples.

umpirsky commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+
+    // Zend\Validator\Iban
+    "Unknown country within the IBAN" => "Непозната земља унутар IBAN-а",
+    "Countries outside the Single Euro Payments Area (SEPA) are not supported" => "Земље изван Single Euro Payments Area (SEPA) нису подржане",
```
Земље изван јединственог подручја плаћања у еврима (SEPA) нису подржане

sttewan replied:
Thanks! It's changed.

umpirsky commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+    "The input does not appear to be a valid IP address" => "Унета је неисправна IP адреса",
+
+    // Zend\Validator\IsInstanceOf
+    "The input is not an instance of '%className%'" => "Унос није инстанца '%className%'",
```
Унос није инстанца класе '%className%'

sttewan replied:
Thanks! It's changed.

nikolaposa commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+    "The input is an empty string" => "Унет је празан текст",
+
+    // Zend\I18n\Validator\Alpha
+    "Invalid type given. String expected" => "Погрешан тип унет. Очекиван је текст",
```
"Pogrešan tip unet" sounds like a too literal translation and it is kinda awkward. "Unet je pogrešan tip" or simply "Pogrešan tip" sounds more natural.

sttewan replied:
All changed to "Унет је погрешан тип". Thanks!

nikolaposa commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+    "The input is an empty string" => "Унет је празан текст",
+
+    // Zend\I18n\Validator\DateTime
+    "Invalid type given. String expected" => "Погрешан тип унет. Очекиван је текст",
```
... same as above.

sttewan replied:
All changed to "Унет је погрешан тип". Thanks!

nikolaposa commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+
+    // Zend\Validator\Callback
+    "The input is not valid" => "Унос је неисправан",
+    "An exception has been raised within the callback" => "Дошло је до грешке приликом позива",
```
IMHO "callback" in this case should be translated more precisely, because this error is about PHP callback function, and not any callback. I would go with something like "Došlo je do greške u toku izvršavanja pozvane funkcije".

nikolaposa replied:
... or simply hard-code "callback", something like: "Дошло је до грешке приликом извршавања 'callback' функције".

sttewan replied:
Changed

nikolaposa commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+    "The input seems to contain an invalid checksum" => "Унос садржи неисправну вредност",
+    "The input must contain only digits" => "Унос мора да садржи само бројеве",
+    "Invalid type given. String expected" => "Погрешан тип унет. Очекиван је текст",
+    "The input contains an invalid amount of digits" => "Унос садржи неисправан број цифара",
```
"неисправан" -> "недозвољен"?

sttewan replied:
Changed

nikolaposa commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+    "An exception has been raised while validating the input" => "Дошло је до грешке приликом провере уноса",
+
+    // Zend\Validator\Csrf
+    "The form submitted did not originate from the expected site" => "Форма није поднета са очекиваног сајта",
```
- "Подаци послати путем форме не потичу са очекиваног сајта"
- "Послати подаци не потичу са очекиваног сајта"

What say?

sttewan replied:
Second is OK. Changed

nikolaposa commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+    "Too few files, minimum '%min%' are expected but '%count%' are given" => "Премало фајлова, очекивани минимум је '%min%' а унет је '%count%'",
+
+    // Zend\Validator\File\Crc32
+    "File does not match the given crc32 hashes" => "Фајл не одговара датим crc32 вредностима",
```
In my opinion, foreign word segments should be wrapped with quotation marks, for example: **Фајл не одговара датим 'crc32' вредностима**

sttewan replied:
I would not add quotations, it will be to much of them in some cases, and they are already in use for variables in messages.

nikolaposa commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+
+    // Zend\Validator\File\IsImage
+    "File is no image, '%type%' detected" => "Фајл није слика, '%type%' детектован",
+    "The mimetype could not be detected from the file" => "Мimеtype фајла не може бити детектован",
```
Quotation marks in case of foreign words: 'Мimеtype' фајла не може бити детектован.

sttewan replied:
Same as the above. Better not to use quotations for every foreign word.

nikolaposa commented on languages/sr_RS/Zend_Validate.php:
```diff
@@ -0,0 +1,308 @@
+    "The input contains non-hexadecimal characters" => "Унос садржи не-хексадецималне карактере",
+
+    // Zend\Validator\Hostname
+    "The input appears to be a DNS hostname but the given punycode notation cannot be decoded" => "Унос је DNS hostname али дату punycode нотацију је немогуће декодирати",
```
Single quotation marks around foreign words?

sttewan replied:
Better not. Thanks!

nikolaposa commented:
I've noticed that in several cases, foreign words that cannot be translated, i.e. md5, hostname, DNS, and similar were left untouched, which is fine, but I suggest that they are surrounded with single quotation marks or such. For example:
- Фајл не одговара датим 'crc32' вредностима
- 'Мimеtype' фајла не може бити детектован.
- Унос је 'DNS hostname' али дату 'punycode' нотацију је немогуће декодирати

Commit:
Update Zend_Validate.php

Updating sr_RS/Zend_Validate.php for Serbian Cyrillic pre-translated validation messages based on collaborative comments.

Commit:
Merge pull request #4 from sttewan/local-dev

Update Zend_Validate.php

sttewan commented:
Above changes committed to local branch.

Maks3w commented:
Any other objection? If not I guess this could be merged along the next days.

weierophinney commented:
Thanks, @sttewan! Merged to develop for release with 2.6.0.
